src/Bearded.TD/Game/Simulation/Elements/ElementalPhenomenonScopeBase.cs

```diff
@@ -1,23 +1,27 @@
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.InteropServices;
 using Bearded.TD.Game.Simulation.GameObjects;
 using Bearded.TD.Game.Simulation.StatusDisplays;
+using Bearded.TD.Utilities.Collections;
 using Bearded.Utilities.SpaceTime;
 
 namespace Bearded.TD.Game.Simulation.Elements;
 
-abstract class ElementalPhenomenonScopeBase<TEffect> : IElementalPhenomenon.IScope<TEffect> where TEffect : IElementalEffect
+abstract class ElementalPhenomenonScopeBase<TEffect>
+    : ElementalPhenomenonScopeBase, IElementalEffect<TEffect>.IScope
+    where TEffect : struct, IElementalEffect<TEffect>, IEquatable<TEffect>
 {
-    private readonly GameObject target;
     private readonly IStatusTracker? statusDisplay;
     private readonly List<EffectWithExpiry> activeEffects = [];
     private ActiveEffect? activeEffect;
 
-    protected IEnumerable<TEffect> ActiveEffects => activeEffects.Select(e => e.Effect);
+    protected GameObject Target { get; private set; }
```

Comment:
What is your reason for making this protected? It feels like we're removing a level of encapsulation here: the scopes that extend this class should really only have access and do things with it in the places we expect them to. That is most lifecycle methods right now, but I could also see an argument to replace that `GameObject` argument with a `EffectApplicationContext` that has methods for adding components and upgrades and even automatically removes them when the effect expires or is replaced with another effect. Then having access to the `Target` even becomes undesirable because it goes outside the normal flow of things, so it feels like this change breaks purposeful encapsulation.